### ISSUE TITLE
Create a factory to replace Schema::make()

### ISSUE THREAD

**Issue by shadowhand** [RoundingWellOS/schematic#7]
https://scrutinizer-ci.com/g/RoundingWellOS/schematic/inspections/495539f2-e137-42ad-9c23-9eafffcca08a/code-structure/operation/RoundingWell%5CSchematic%5CSchema%3A%3Amake

Scruitinizer doesn't like all the conditions in `Schema::make`. The current implementation is also ~~hard~~ impossible to extend cleanly. Having a factory with a list of types would work better and should solve the the quality issue.

**Comment by coderabbi** [comment]
A switch will shut scrutinizer up. 🤣 

A simple, though ever so slightly magical, approach would be:

``` 

 protected const SCHEMA_TYPES = [
        'array',
        'boolean',
        'integer',
        'null',
        'number',
        'object',
        'string'
 ];

 public static function make($json): Schema
 {
        if ( ! isset($json->type)) {
            throw new \InvalidArgumentException('Missing schema type.');
        }

        if ( ! in_array(strtolower($json->type), Schema::SCHEMA_TYPES)) {
            throw new \InvalidArgumentException(sprintf(
                "No schema type available for %s.",
                $json->type
            ));
        }

        $schema = 'Schema\\' . ucfirst($json->title) . 'Schema';

        return new $schema();
 }
```

Thoughts?

**Comment by coderabbi** [comment]
Let's see what scrutinizer does with it...

**Comment by shadowhand** [comment]
Fixed by #8.
